src/eradius_proxy.erl

```diff
@@ -39,25 +50,25 @@
                           {timeout, ?DEFAULT_TIMEOUT},
                           {retries, ?DEFAULT_RETRIES}]).
 
--type route() :: eradius_client:nas_address().
--type routes() :: [{Name :: string(), route()}].
+-type route() :: {default_route, eradius_client:nas_address()}.
+-type routes() :: {routes, [{Name :: string(), route()}]}.
 -type undefined_route() :: {undefined, 0, []}.
 
 radius_request(Request, _NasProp, Args) ->
-    DefaultRoute = proplists:get_value(default_route, Args),
+    DefaultRoute = get_proxy_opt(default_route, Args),
+    Routes = get_proxy_opt(routes, Args),
     Options = proplists:get_value(options, Args, ?DEFAULT_OPTIONS),
     Username = eradius_lib:get_attr(Request, ?User_Name),
-    Routes = proplists:get_value(routes, Args, []),
     {NewUsername, Route} = resolve_routes(Username, DefaultRoute, Routes, Options),
     Retries = proplists:get_value(retries, Options, ?DEFAULT_RETRIES),
     Timeout = proplists:get_value(timeout, Options, ?DEFAULT_TIMEOUT),
     SendOpts = [{retries, Retries}, {timeout, Timeout}],
     send_to_server(new_request(Request, Username, NewUsername), Route, SendOpts).
 
 validate_arguments(Args) ->
-    DefaultRoute = proplists:get_value(default_route, Args, {undefined, 0, []}),
+    DefaultRoute = lists:keyfind(default_route, 1, Args),
     Options = proplists:get_value(options, Args, ?DEFAULT_OPTIONS),
-    Routes = proplists:get_value(routes, Args, []),
+    Routes = lists:keyfind(routes, 1, Args),
     case {validate_route(DefaultRoute), validate_options(Options), compile_routes(Routes)} of
```

Comment:
Is this case finding correct/incorrect default_route, options, route? It would be clearer if the return values where
incorrect_default_route, incorrect_options, incorrect_routes.
When correct only the list need to be returned. It is different enough from the incorrect.

Reply:
@ebengt 

> Is this case finding correct/incorrect default_route, options, route

yes, it is this. Now in a case of incorect default route, routes or options we already return atom `default_route`, `options` or `routes` and this atom will be used in error logs here - https://github.com/travelping/eradius/blob/master/src/eradius_config.erl#L79, so it should be pretty clear in a case of such errors.